modules/connectivity/ParticipantConnectionStatus.js

```diff
@@ -0,0 +1,306 @@
+/* global __filename, module, require */
+var logger = require('jitsi-meet-logger').getLogger(__filename);
+var MediaType = require('../../service/RTC/MediaType');
+var RTCBrowserType = require('../RTC/RTCBrowserType');
+var RTCEvents = require('../../service/RTC/RTCEvents');
+
+import * as JitsiConferenceEvents from '../../JitsiConferenceEvents';
+import * as JitsiTrackEvents from '../../JitsiTrackEvents';
+
+/**
+ * How long we're going to wait after the RTC video track muted event for
+ * the corresponding signalling mute event, before the connection interrupted
+ * is fired.
+ *
+ * @type {number} amount of time in milliseconds
+ */
+const RTC_MUTE_TIMEOUT = 1000;
+
+/**
+ * Class is responsible for emitting
+ * JitsiConferenceEvents.PARTICIPANT_CONN_STATUS_CHANGED events.
+ *
+ * @constructor
+ * @param rtc {RTC} the RTC service instance
+ * @param conference {JitsiConference} parent conference instance
+ */
+function ParticipantConnectionStatus(rtc, conference) {
```

Comment:
It's going to be better if we implement this class with the ES6 syntax since we support it now.